src/topology/algebra/order/basic.lean

```diff
@@ -1157,6 +1157,96 @@ lemma filter.eventually.exists_Ioo_subset [no_max_order α] [no_min_order α] {a
   ∃ l u, a ∈ Ioo l u ∧ Ioo l u ⊆ {x | p x} :=
 mem_nhds_iff_exists_Ioo_subset.1 hp
 
+/-- The set of points which are isolated on the right is countable when the space is
+second-countable. -/
+lemma countable_of_isolated_right [second_countable_topology α] :
+  set.countable {x : α | ∃ y, x < y ∧ Ioo x y = ∅} :=
+begin
+  casesI is_empty_or_nonempty α,
```

Comment:
I would prefer to add `@[nontriviality]` lemmas `set.finite_of_subsingleton` and `set.countable_of_subsingleton`. Otherwise LGTM.
bors d+